src/stactools/viirs/metadata.py

```diff
@@ -388,4 +392,8 @@ def viirs_metadata(
     if href_exists(read_xml_href):
         return Metadata.from_xml_and_h5(h5_href, read_href_modifier, densify_factor)
     else:
-        return Metadata.from_h5(h5_href, read_href_modifier, densify_factor)
+        product = os.path.basename(xml_href).split(".")[0]
```

Comment:
Curious if we do product-from-file-name-parsing anywhere else in the repo. If so, we should DRY this.

Reply:
Actually, we don't. Made a utility function for it anyway, though.